satpy/readers/msi_safe.py

```diff
@@ -201,7 +201,7 @@ def _get_coarse_dataset(self, key, info):
             return
```

Comment:
Could you remove this unnecessary else statement while you are working on this file?

Reply:
I've left the `else` as I think it makes things clearer but have now explicitly added `return None` instead of just `return`. Same in the function below as well.